### ISSUE TITLE
ENH: Improve stacked coordinate interface

### ISSUE THREAD

**Issue by mpu-creare** [creare-com/podpac#339]
We should add the option to do the following:

```python
c = Coordinates([[lat, lon], time], dims=[['lat', 'lon'], 'time'])
```
Instead, we currently have:
```python
c = Coordinates([[lat, lon], time], dims=['lat_lon', 'time'])
```

**Comment by jmilloy** [comment]
We use `lat_lon` for StackedCoordinates and `lat,lon` for DependentCoordinates. Eventually, these should be merged into a single class (see #268). For now, I am choosing that `dims=[['lat', 'lon'], 'time]` means StackedCoordinates. I don't think that will be a problem.

**Comment by mpu-creare** [comment]
see 84019ab737cb9c76a11ee598f21dccc8fe10c6cd
